### DISCUSSION TITLE
max_in_flight option

### DISCUSSION THREAD

**Discussion by Nowogrodzki** [vernemq/vernemq#2165] [Q&A] [answered]
Hello VerneMQ Team!

 I have a question regarding the max_in_flight option.
When I specified max_in_flight=100 is it means that 100 messages will be sent to each topic/subscription or is it 100 messages for an entire client

**Comment by ioolkos** [comment, marked as answer]
@Nowogrodzki `max_inflight_messages` is applied per client session, at the publishing end of the broker. That is: between the broker and 1 MQTT client. This is not related to topics.

Note that this also only applies to protocol version before 5. For similar behaviour with MQTT v5, look into `receive_max_client`/`receive_max_broker`

**Reply by Nowogrodzki** [reply]
@ioolkos Thank you so much for the answer! 

I'm using MQTT v5, looking at these two options mentioned you would recommend setting the same values for both? Also, should I remove `max_in_flight` option due to using the v5 version?

**Reply by ioolkos** [reply]
@Nowogrodzki as stated, `max_inflight_messages` will not apply to v5. Whether you have v3 clients or not, I cannot know. The default will be 20 anyway, whether you configure it or not.

To understand Receive Max for v5: https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901049

Verne settings:
`receive_max_client`: MQTT clients are free to choose their own ReceiveMax, but Verne will apply `receive_max_client` as an upper limit.
`receive_max_broker`: if the Client can state that it cannot do more than ReceiveMax, then the broker can too. `receive_max_broker` is to configure the broker to say "I cannot do more than X messages unacklowledged". If a Client comes in with a higher value, Verne will state its own value in the `CONNACK` answer but then disconnect the client.

Yeah, I'd probably configure at least `receive_max_client` to something a little lower than 65k.

**Reply by Nowogrodzki** [reply]
@ioolkos Thank you very much for that information :) I have one more question, but this time regarding shared subscriptions, In the 1.12.4 version new policy has been added, namely `prefre_online_before_local`, but I saw in the one discussion that you wrote about a policy named `prefer_online_to_local`, is that the same policy but in the debate or changelog there was a naming issue?

**Reply by ioolkos** [reply]
@Nowogrodzki if the inflight question is answered, can you mark it answered?
I do not understand the question about shared subscriptions. Yes, typos are possible.

**Reply by Nowogrodzki** [reply]
Marked :) Regarding shared subscription policies, which name is correct `prefer_online_before_local` or `prefer_online_to_local`
In the changelog under the 1.12.4 version new policy is named `prefer_online_before_local`, In this discussion [here](https://github.com/vernemq/vernemq/discussions/1959) you wrote about `prefer_online_to_local`, I'm a little bit confused about which one is correct

**Reply by ioolkos** [reply]
ah, sorry, yes it's `prefer_online_before_local`.
